### ISSUE TITLE
TypeError: val.indexOf is not a function on entering value

### ISSUE THREAD

**Issue by Carlitta87** [s-yadav/react-number-format#259]
I am currently using Material-UI/core in react. I created a custom input field where I could place anywhere. As I entered a value, I received this error.
![image](https://user-images.githubusercontent.com/2539915/49756828-e440bd80-fc88-11e8-84ed-b04e1bf9ff84.png)

**Comment by JosipTomaic** [comment]
@Carlitta87 Hello, did you solve this issue? I'm having identical problem...

**Comment by namxam** [comment]
Yeah, me too. It happens when I manage the value in parent component and pass it down to NumberFormat component.

**Comment by TrevorJamesH** [comment]
It looks like this happens if you pass null as the value. Check for null somehow and pass an empty string instead.

**Comment by slim-hmidi** [comment]
Does anyone fixes this problem?

**Comment by yashcr07** [comment]
In the `onChange` function we receive a value of object type. See [here](https://github.com/s-yadav/react-number-format/blob/master/README.md#values-object)
so you would need to access the `formattedValue` key in that object and use that to set the value.
I hope it helps

**Comment by caretapps-developer** [comment]
@yashcr07 i tried what you mentioned and I see the 3 values. However he issue persists.  Did anyone resolve this issue successfully?

**Comment by Joy-jeevan** [comment]
solved this issue by setting isNumericString props to true. see [here](https://www.npmjs.com/package/react-number-format#notes-and-quirks)

**Comment by Badlapje** [comment]
I have this same issue, not solved by any of the above.  Interesting enough, it doesn't happen if i call NumberFormat directly.  It does happen if i create a wrapper for it like so: 
```js
export default function EuroFormat(amount) {
  return <NumberFormat decimalSeparator="," displayType="text" prefix="€ " thousandSeparator="." value={amount} />
}
```

Not sure why this happens, but i reckon it shouldn't.
